docs/UnitTestBarAndStandards.md

```diff
@@ -0,0 +1,23 @@
+<!-- Copyright (c) Microsoft Corporation. All rights reserved.
+     Licensed under the MIT License. -->
+     
+     ## Test requirements
+* All new non-UI code must have unit tests.
+* Unit tests need to accompany the code in the pull request. This provides implicit documentation of the expected behavior, and ensures that the behavior is being satisfied. In an exceptional circumstance, a PR might be approved without its unit tests, but that is not the normal expectation.
+* All unit tests should be passing locally before the PR is created.
+* Unit tests must conform to the standards listed below.
+
+## Standards
+* The test project is appropriately named (tests for the `Foo.Bar.dll` are in `Foo.BarTests.dll`)
+* The test namespace is appropriately named (the top level namespace matches the assembly name, then the hierarchy in the assembly is preserved. So if `Foo.Bar.dll` contains a namespace named `Widget`, the tests will be in the `Foo.BarTests.Widget` namespace)
+* The test class is appropriately named (tests for the `ExplodingWidget` class will be in the `ExplodingWidgetTests` class)
+* Each test follows the naming conventions documented at [Unit Test Basics](https://msdn.microsoft.com/en-us/library/hh694602.aspx) Summarizing:
+  * Name is in the general form `MethodName_TestCase_ExpectedBehavior`
+  * If the test is expected to throw an `Exception`, the test should declare this via the `ExpectedException` attribute. If the test needs to check the contents of the `Exception`, the best pattern is to catch the expected `Exception` type, validate it, then re-throw the `Exception`
+* Each test uses appropriate `Assert` methods to validate its conditions--use the version with a hint where it seems helpful
+* Each test specifies a reasonable `Timeout` value (default to 1 second each). The test output is much more readable if the test times out and fails than if the entire build loop hangs and reaches the 60 minute time constraint
+  * One downside to the `Timeout` attribute--because the `Timeout` applies while debugging the unit test, it may need to be commented out to make debugging feasible.
+* Appropriate mocks or fakes decouple the test class from the underlying dependencies (Dependency Injection is a common pattern for this). This is required for new classes, best effort for changes to existing classes
+* `Microsoft Fakes` may not be used for any tests because not all editions of Visual Studio support them.
```

Comment:
>`Microsoft Fakes` may not be used for any tests [](start = 2, length = 47)

This line may need some qualifiers since we currently have unit tests that use Fakes. Maybe exclude it from new tests?

Reply:
```suggestion
* `Microsoft Fakes` may not be used for any new tests because not all editions of Visual Studio support them.
```